packages/controllers/src/permissions/PermissionController.test.ts

```diff
@@ -0,0 +1,112 @@
+import { ControllerMessenger, Json } from '@metamask/controllers';
+import {
+  AsyncRestrictedMethodImplementation,
+  CaveatDecorator,
+  CaveatSpecifications,
+  PermissionController,
+  PermissionControllerActions,
+  PermissionControllerEvents,
+  PermissionEnforcer,
+  PermissionSpecifications,
+  RestrictedMethodArgs,
+} from '.';
+
+const controllerName = 'PermissionController' as const;
+
+function getRestrictedMessenger() {
+  const controllerMessenger = new ControllerMessenger<
+    PermissionControllerActions,
+    PermissionControllerEvents
+  >();
+  return controllerMessenger.getRestricted<typeof controllerName, never, never>(
+    {
+      name: controllerName,
+    },
+  );
+}
+
+enum CaveatTypes {
+  filterArrayResponse = 'filterArrayResponse',
+}
+
+function getCaveatSpecifications(): CaveatSpecifications {
+  return {
+    filterArrayResponse: {
+      type: CaveatTypes.filterArrayResponse,
+      decorator: ((
+          method: AsyncRestrictedMethodImplementation<Json, string[]>,
+          caveat: { type: CaveatTypes.filterArrayResponse; value: string[] },
+        ) =>
+        async (args: RestrictedMethodArgs<Json>) => {
+          const result = await method(args);
+          return Array.isArray(result)
+            ? result.filter(caveat.value.includes)
+            : result;
```

Comment:
Defaulting to returning an unfiltered response seems dangerous :thinking: 

As an example caveat it's fine of course. But throwing an error here might be more representative of how a real caveat would/should behave. Otherwise you could bypass the caveat by having the result not be of the expected type (e.g. a custom array class might fail `Array.isArray` but still work as an array)